temporal-test-server/src/test/java/io/temporal/testserver/functional/MultiOperationTest.java

```diff
@@ -0,0 +1,446 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.testserver.functional;
+
+import static org.junit.Assert.*;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.common.v1.WorkflowType;
+import io.temporal.api.enums.v1.TaskQueueKind;
+import io.temporal.api.enums.v1.UpdateWorkflowExecutionLifecycleStage;
+import io.temporal.api.errordetails.v1.MultiOperationExecutionFailure;
+import io.temporal.api.taskqueue.v1.TaskQueue;
+import io.temporal.api.update.v1.Input;
+import io.temporal.api.update.v1.Meta;
+import io.temporal.api.update.v1.Request;
+import io.temporal.api.update.v1.WaitPolicy;
+import io.temporal.api.workflowservice.v1.*;
+import io.temporal.client.*;
+import io.temporal.common.converter.DefaultDataConverter;
+import io.temporal.serviceclient.StatusUtils;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.testserver.functional.common.TestWorkflows;
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MultiOperationTest {
```

Comment:
Can we verify the multi op API timeout also works properly? I don't see any test coverage here and in the implementation I don't think the underlying update request is being called with a deadline? You can see the test coverage we have for regular update like `updateNotAcceptedTimeout`

Reply:
Okay, I'm writing that test right now and it seems that the retries I added are causing ALREADY_EXISTS errors. From what I can tell right now, it seems like the Test Server is not deduping start requests based on request ID. I'm might have to fix that first.

Reply:
Okay, I'm having an issue here: when mimicking the `updateNotAcceptedTimeout` test for MultiOps, I see a `DEADLINE_EXCEEDED: deadline exceeded after 9.999713584s.` client-side error. I can also see in the debugger that the server attempted to respond, but the client timeout throws first.

I couldn't yet figure out how to bump the client-side timeout up. I tried `.withDeadlineAfter(30, TimeUnit.SECONDS)` but it has no effect, it still times out after ~10s.

Reply:
@Quinn-With-Two-Ns would you know what's going on here?

Reply:
Addressed in https://github.com/temporalio/sdk-java/pull/2199/commits/df7a43d285bd050101d835e6f1a5eae56bee49e4